src/gui/plugins/view_angle/ViewAngle.cc

```diff
@@ -61,7 +76,17 @@ void ViewAngle::LoadConfig(const tinyxml2::XMLElement *)
     this->title = "View Angle";
 
   // For view angle requests
-  this->dataPtr->service = "/gui/view_angle";
+  this->dataPtr->viewAngleService = "/gui/view_angle";
+
+  // Subscribe to camera pose
+  std::string topic = "/gui/camera/pose";
+  this->dataPtr->node.Subscribe(
+    topic, &ViewAnglePrivate::CamPoseCb, this->dataPtr.get());
+
+  // Move to pose service
+  this->dataPtr->moveToPoseService = "/gui/move_to/pose";
+
+  this->dataPtr->viewAngle = this;
```

Comment:
This is adding a circular dependency which could bring problems down the road. How about adding the `CamPoseCb` function to `ViewAngle` instead of `ViewAnglePrivate` so that you can do `this->CamPoseChanged()`?

Reply:
ahhh I had it that way because I misremembered what broke abi :sweat_smile: fixed 2b4c28a5b5178962e6721685c2171f5709629c52

Reply:
Ah gotcha. For future reference, we don't need to worry about ABI for Gazebo-GUI plugins, because their headers are not being installed. We're using the PIMPL pattern on some of them just in case we ever need to install the headers.